frontend/src/components/Modal.tsx

```diff
@@ -0,0 +1,61 @@
+import React, { useEffect } from 'react';
+import ReactDOM from 'react-dom';
+import { useModalStack } from '@/context/ModalContext';
+import { useModal } from '@/hooks/useModal';
+
+type ModalProps = {
+  children: React.ReactNode;
+  id: string;
+  isOpen: boolean;
+  onClose: () => void;
+};
+
+export const Modal: React.FC<ModalProps> = ({
+  id,
+  isOpen,
+  onClose,
+  children,
+}) => {
+  // eslint-disable-next-line react/hook-use-state
+  const modalRef = useModal(isOpen, onClose);
+  const { modalStack, pushModal, popModal } = useModalStack();
+
+  useEffect(() => {
+    // eslint-disable-next-line no-debugger
+    if (isOpen) {
+      pushModal(id);
+    } else {
+      popModal();
+    }
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, [isOpen, id]);
```

Comment:
here there are some pending deps, but the react render cycle was getting out of hand, so i had to limit it to what was relevant, again i think some strict rules should be ease in this kinda of case, given that we should add direct deps